src/services/utils.js

```diff
@@ -0,0 +1,122 @@
+import dayjs from 'dayjs'
+
+const PERIOD = 'DAILY'
+// Percentage for color gradient according to the mean price for the last 7 days
+const PERCENTAGE_MEAN = 10 / 100 // 10%
+
+export function transformIndexedTariffPrices(firstDate, calendarDay, prices) {
+  const first_date = new Date(firstDate)
+  first_date.setHours(0)
+  const calendar_day = new Date(calendarDay)
+  const [startIndex, endIndex] = sliceIndexes(first_date, PERIOD, dayjs(calendar_day))
+
+  var measured_data = timeSlice(first_date, prices, startIndex, endIndex)
+
+  let acum = prices.reduce((accumulator, currentValue) => {
+    return accumulator + currentValue
+  }, 0)
+  let average = acum / prices.length
+
+  const today = new Date()
```

Comment:
This today should be calendarDay,because if you use the current date as the reference day won't be correct in the moment the user change the day in the datepisker